automations/python/label_pr.py

```diff
@@ -220,11 +223,12 @@ def main():
                 labels_to_add.append(label)
 
         if labels_to_add:
-            pr.set_labels(*labels_to_add)
+            pr.set_labels(*labels_to_add, *pr.labels)
             # Only break when all labels are applied, if we're missing any
             # then continue to the else to apply the awaiting triage label.
-            # Stack can have more than one label so this is not an exact check
-            if len(labels_to_add) >= len(required_label_categories):
+            # Stack can have more than one label and is typically already applied
+            # by this step, so remove it from the total count of labels to add.
```

Comment:
I didn't quite get what "typically already applied by this step" refers to. Did you mean that the CI + CD step to apply stack labels would be reached first if both the CI + CD and labelling workflows were started simultaneously by GitHub?

Reply:
Yes, that's exactly what I mean! I'll reword this 🙂